kernel/src/managers/dma/dma.c

```diff
@@ -384,8 +435,68 @@ kstatus_t mgr_dma_stream_start(const dmah_t dmah)
  *   K_ERROR_BADSTATE: the stream is not started
  *   K_STATUS_OKAY: stream has been stopped
  */
-kstatus_t mgr_dma_stream_stop(const dmah_t dmah)
+kstatus_t mgr_dma_stream_suspend(const dmah_t dmah)
 {
     kstatus_t status = K_ERROR_INVPARAM;
+    dma_stream_config_t * const cfg = mgr_dma_get_config(dmah);
+
+    if (unlikely(cfg == NULL)) {
+        goto end;
+    }
+    /* config entry, when found, must have its meta field properly set (dma_init time set) */
+    /*@ assert \valid_read(cfg); */
+    /*@ assert \valid_read(cfg->meta); */
+
+    /* can't unassign a stream that is started or already unassigned */
+    if (unlikely(
```

Comment:
weird format

Reply:
right... old check not cleanly removed. fixed.